CodeU_Assignement_3/test/codeu_assignement_3/KemWordSearchTest.java

```diff
@@ -0,0 +1,145 @@
+        String[] dict_word = {"CAR", "CARD", "CART", "CAT"};
+        SimpleDictionary dict = new SimpleDictionary(dict_word);
+        KemWordSearch instance = new KemWordSearch();
+        String[] expResult = null;
+        String[] result = instance.wordSearch(row, col, grid,dict);
+        assertArrayEquals(expResult, result);
+    }
+
+    /**
+     * Test of wordSearch method, of class KemWordSearch.
+     */
+    @Test
+    public void testWordSearchEmptyDict() {
+        System.out.println("wordSearch Empty Dictionary");
+        int row = 0;
+        int col = 0;
+        char[][] grid = {{'a','a','r'},{'t','c','d'}};
+        SimpleDictionary dict = new SimpleDictionary(null);
+        KemWordSearch instance = new KemWordSearch();
+        String[] expResult = null;
+        String[] result = instance.wordSearch(row, col, grid,dict);
+        assertArrayEquals(expResult, result);
+    }   
+    
+    /**
+     * Test of wordSearch method, of class KemWordSearch.
+     */
+    @Test
+    public void testWordSearch() {
+        System.out.println("wordSearch wiht Dictionary");
+        int row = 2;
+        int col = 3;
+        char[][] grid = {{'A','A','R'},{'T','C','D'}};
+        String[] dict_word = {"CAR", "CARD", "CART", "CAT"};
+        SimpleDictionary dict = new SimpleDictionary(dict_word);
+        KemWordSearch instance = new KemWordSearch();
+        String[] expResult = {"CAR", "CARD", "CAT"};
+        String[] result = instance.wordSearch(row, col, grid,dict);
+        assertArrayEquals(expResult, result);
+    }
+    
+    /**
+     * Test of wordSearch method, of class KemWordSearch.
+     */
+    @Test
+    public void testWordSearch2() {
+        System.out.println("wordSearch with Dictionary (2)");
+        int row = 3;
+        int col = 2;
+        char[][] grid = {{'A','T'},{'A','C'},{'R','D'}};
+        String[] dict_word = {"CAR", "CARD", "CART", "CAT","AT","FACE","COOL","OPEN","RAT"};
+        SimpleDictionary dict = new SimpleDictionary(dict_word);
+        KemWordSearch instance = new KemWordSearch();
+        String[] expResult = {"AT","CAR", "CARD", "CAT","RAT"};
+        String[] result = instance.wordSearch(row, col, grid,dict);
+        assertArrayEquals(expResult, result);
+    }
+    
+    /**
+     * Test of wordSearch method, of class KemWordSearch.
+     */
+    @Test
+    public void testWordSearch3() {
+        System.out.println("wordSearch with Dictionary (3)");
+        int row = 3;
+        int col = 3;
+        char[][] grid = {{'A','T','I'},{'A','C','J'},{'R','D','K'}};
+        String[] dict_word = {"CAR", "CARD", "CART", "CAT","AT","FACE","COOL","OPEN","RAT","I","ARC"};
+        SimpleDictionary dict = new SimpleDictionary(dict_word);
+        KemWordSearch instance = new KemWordSearch();
+        String[] expResult = {"AT","I","ARC","CAR", "CARD", "CAT","RAT"};
+        String[] result = instance.wordSearch(row, col, grid,dict);
+        assertArrayEquals(expResult, result);
+    }
+}

```

Comment:
There are some good tests here but, as with the other file, it's written in an unnecessarily verbose style. Look at your colleagues' code for ideas on how to make the tests shorter.

I don't see any code that tests for repeated use of characters - you could have a bug where the same character is used twice, and i don't think these tests would cover it.